crates/ruff_linter/src/rules/ruff/rules/missing_fstring_syntax.rs

```diff
@@ -209,7 +209,15 @@ fn should_be_fstring(
                     return false;
                 }
                 if semantic
-                    .lookup_symbol(id)
+                    // the parsed expression nodes have incorrect ranges
+                    // so we create a dummy node which has a range that's
+                    // close enough to being correct in order for
+                    // `simulate_runtime_load` to return the correct result.
+                    .simulate_runtime_load(&ast::ExprName {
+                        id: id.clone(),
+                        range: literal.range(),
+                        ctx: ast::ExprContext::Load,
+                    })
```

Comment:
I think I have a slight preference to adding an extra method here that takes a string and range over creating a fake AST
```suggestion
                    .simulate_runtime_load_str(&id, literal.range())
```

Reply:
I considered that as well. My most favorite solution would be to extend `Parser` with an optional static offset, so the parsed nodes get created with the correct offset.